.github/workflow/prettier.yml

```
@@ -0,0 +1,25 @@
+name: Continuous Integration
+
+# This action works with pull requests and pushes
+on:
+  pull_request:
+  push:
+    branches:
+      - main
```

To see it in action could you temporarily set up to 
```suggestion
  pull_request:
    branches:
      - main
  push:
    branches:
      - main
```
See (after the commit) should force the action to be ran on this PR (we will see the output in the Actions tab
And we'll be able (maybe?) to understand its output (in order to see how it handles the commiting part of the formatting :+1: )

---

I can do it if you want, but I tested it on another repo and it works perfectly
The output (if a none formatted file is found) is something like
```
[file.js] Error: Did you forget to run prettier on this file?
```

---

Ok that's wonderful but does it tell the user what the error is?
Maybe somebody won't want to install prettier globally (and currently we don't have a dev env, and #64 won't introduce such possibilities I think)

---

You mean the error should give a way to fix the formatting?
Mmm if one of the contributors doesn't want to install npm then it's a little bit tricky to agree with the formatting.
You can install it locally if you don't want it globally 🤔 
And I can definitely make some doc if needed.

---

so please do not change the `on:` field @atxr.

it should run as-is :+1:

---

> so please do not change the on: field @atxr.

@amtoine 
You actually got me wrong here... It wasn't a request to change the `on:` field *(even if I believe that running this `on: pull_request: main` would be a good idea and would have avoided the confusing discussion I see in the comments below...)*

My point here is that the error msg would be far better if it also showed where the formatting error are. As pylint and black do for example!

----------------------

>  Mmm if one of the contributors doesn't want to install npm then it's a little bit tricky to agree with the formatting.

@atxr I quite don't agree with you here... Actually the dev env I'm building in #64 and #86 is designed to avoid installing iscsc.fr dev stuff on your local machine! I see `npm` and `prettier` as such dev tools so maybe I'll write a little script that runs the docker containers and run prettier in it, it might seems over-complicated to you but I like not to mess with my machine, and I'm creating the dev env for this.

>  You can install it locally if you don't want it globally thinking

And this isn't a suited solution as we decided together a few months ago not to install packages locally in root, installing prettier locally would modify the package.json and create a package-lock.json at root :+1:

---

> My point here is that the error msg would be far better if it also showed where the formatting error are. As pylint and black do for example!

I don't know if such a feature exists, I'll check for you but I didn't find anything like this

> I see `npm` and `prettier` as such dev tools so maybe I'll write a little script that runs the docker containers and run prettier in it, it might seems over-complicated to you but I like not to mess with my machine, and I'm creating the dev env for this.

If you think it's a great move to add it the the dev env, it can be great. However, I don't think we can change anything about this in the scope of this PR

> After looking at #84 I remove this statement, but still some people (like me)don't want to install `npm` at all if possible :+1: 
Again, I totally understand but I think it's related with the dev workflow.

---

it's true that, for instance, `cargo fmt --check` shows where the errors are, which is very nice :+1: 

and for the `npm` installation, i agree with you @atxr => this website is `npm` / `node`-based, i.e. you'll need to have a minimal dev setup.
for a `rust` project, you'd have to install `rustc` and `cargo`.
for a `python` project, you'd have to install `python` and probably `pylint` or `mypy`.
etc, etc...

---

for me, the preview of the changes to be applied is nice, but not that critical.
and that can be addressed in another PR :+1:

---

Again, I understand that one doesn't want to install those tools on a local setup, because of conflicts for exemple 🙄
But I think we should merge this PR and add improvement to the dev workflow to fix this

---

agree, let's land this and build upon it :+1: 

@ctmbl are you ok with that?
you can resolve this and merge the PR :ok_hand:

---

ooh no, @atxr please solve the conflict :wink:

---

Yep I agree with both of you these two points:
 - show format error diff in the prettier action logs
 - wrap prettier in the dev env

should be addressed in other PRs
But I wanted to reach a common opinion about these before resolving this thread!